pkg/odo/cli/component/create.go

```diff
@@ -353,8 +389,20 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			return errors.New("This directory already contains a devfile, you can't specify devfile via --devfile")
 		}
 
+		var catalogList catalog.ComponentTypeList
+		if co.forceS2i {
+			client := co.Client
+			catalogList, err = catalog.ListComponents(client)
+			if err != nil {
+				return err
+			}
+		}
+
 		// Validate user specify devfile path
 		if co.devfileMetadata.devfilePath.value != "" {
+			if co.forceS2i {
+				return errors.New("You can't set --s2i flag as true if you want to use the devfile that is specified via --devfile")
```

Comment:
can we check for all conflicting flags like `--registry, --starter, --token etc` at one place only.  I think this also could be done in the `checkConflictingFlags()` function above.
```
if co.forceS2i {
     if co.devfileMetadata.devfilePath.value  != ""  || co.devfileMetadata.devfileRegistry.Name != ""{ 
     return errors.New("You can't set --s2i flag as true if you want to use the devfile that is specified via --devfile")
}
```

Reply:
The `checkConflictingFlags( )`  is actually there for the clause 
` if the user provides a flag that is valid only s2i components without --s2i the command should error out`

from https://github.com/openshift/odo/issues/3403#issue-643752720 .

Maybe I should rename it to checkConfictingS2IFlags

Reply:
what i meant here you could check devfile and s2i conflicting flags at one place only, may be define another fuction `checkS2iDevfileConflictFlags`, i see you are checking each one at separate places.

Reply:
I thought its easier to follow the code flow this way.  https://en.wikipedia.org/wiki/KISS_principle